// safe-units/dist/measure/numberMeasure.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var genericMeasureFactory_1 = require("./genericMeasureFactory");
var genericMeasureUtils_1 = require("./genericMeasureUtils");
var staticMethods = {
    abs: genericMeasureUtils_1.wrapUnaryFn(Math.abs),
    ceil: genericMeasureUtils_1.wrapUnaryFn(Math.ceil),
    floor: genericMeasureUtils_1.wrapUnaryFn(Math.floor),
    fround: genericMeasureUtils_1.wrapUnaryFn(Math.fround),
    round: genericMeasureUtils_1.wrapUnaryFn(Math.round),
    trunc: genericMeasureUtils_1.wrapUnaryFn(Math.trunc),
    hypot: genericMeasureUtils_1.wrapSpreadFn(Math.hypot),
    sqrt: genericMeasureUtils_1.wrapRootFn(Math.sqrt, "2"),
    cbrt: genericMeasureUtils_1.wrapRootFn(Math.cbrt, "3"),
};
var numericOps = {
    one: function () { return 1; },
    neg: function (x) { return -x; },
    add: function (x, y) { return x + y; },
    sub: function (x, y) { return x - y; },
    mult: function (x, y) { return x * y; },
    div: function (x, y) { return x / y; },
    pow: function (x, y) { return Math.pow(x, y); },
    compare: function (x, y) { return x - y; },
    format: function (x) { return "" + x; },
};
exports.Measure = genericMeasureFactory_1.createMeasureType(numericOps, staticMethods);
